src/model.rs

```diff
@@ -159,6 +160,24 @@ impl Collections {
             }
         }
 
+        fn dedup_collection<T: Hash>(source: &mut Collection<T>) {
+            let calculate_hash = |t: &T| -> u64 {
+                let mut s = DefaultHasher::new();
+                t.hash(&mut s);
+                s.finish()
+            };
+            let mut objects_hashes: Vec<u64> = Vec::new();
+            source.retain(|object| {
+                let object_hash = calculate_hash(&object);
+                if objects_hashes.contains(&object_hash) {
+                    false
+                } else {
+                    objects_hashes.push(object_hash);
+                    true
+                }
+            });
+        }
```

Comment:
Something was bothering me about this function, but I couldn't find out why... I think I finally found it. Why using `Hash` where we could simply use `PartialEq`?

```rust
fn dedup_collection<T: PartialEq>(source: &mut Collection<T>) {
    let mut dedup = Vec::default();
    for object in source.take() {
        if !dedup.contains(&object) {
            dedup.push(object);
        }
    }
    *source = Collection::new(dedup);
}
```

That also means that we're going to implement `PartialEq` on `Transfer` only on `from_stop_id` and `to_stop_id` but I'd say it is as good as doing it for the `Hash` (I don't see why we would do it for `Hash` but not `PartialEq`).

Reply:
:+1: 
It should be better as you suggest yes. I also felt a weird thing in this use of hash. I will test this.

Reply:
Done :heavy_check_mark:

Reply:
OK, so it seems it was a bad idea afterall :facepalm: (see https://github.com/CanalTP/transit_model/pull/808).